http-clients/netty-nio-client/pom.xml

```diff
@@ -85,6 +85,14 @@
             <groupId>io.netty</groupId>
             <artifactId>netty-transport-classes-epoll</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.netty</groupId>
+            <artifactId>netty-resolver</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.netty</groupId>
+            <artifactId>netty-resolver-dns</artifactId>
+        </dependency>
```

Comment:
This is an extra 161 KB, for a feature that's off by default. 

It should probably be `optional` for now, with updated javadoc to specify that it requires an extra dependency on `netty-resolver-dns`. Can we also update the error message they get if they try to enable it without the extra dependency to something like "hey, you forgot to depend on io.netty:netty-resolver-dns"? Prior art: https://github.com/aws/aws-sdk-java-v2/blob/7d302ee48653482b71ab3e650c1b2edb6aa0b4c6/core/auth/src/main/java/software/amazon/awssdk/auth/signer/SignerLoader.java#L56-L58

Reply:
While implementing this, I stumped upon a decision: I want to reuse the `ClassLoaderHelper.java` class used in SignerLoader and other classes to dynamically instantiate dependencies. The problem is that this loader resides in the **sdk-core** module. 

On the other hand, **netty-nio-client** does not declare this sdk module. Now I don't now if it's worth this dependency, if it is heavier than 161 KB then it beats the purpouse. I don't know how heavy it is, should I include it? 

Other solution I thought would be just duplicate a small portion of `ClassLoaderHelper.java` inside the **netty-nio-client**, just the necessary to load classes.

Reply:
Hmm, we certainly can't add the dependency, since it would create a cyclic relationship. Honestly, the `auth` module shouldn't even be depending on `ClassLoaderHelper`, since it's a module-internal API. 

Can you create `ClassLoaderHelper` in the `utils` module and use it from there? 

We should go back and move all uses of `ClassLoaderHelper` from the `sdk-core` copy to the `utils` module, and deprecate-for-removal the one in `sdk-core`. I won't ask you to do the moving-everything-else part of it, since that seems like it should be a separate PR.

Reply:
Fair enough, I can create create `ClassLoaderHelper` within utils. 

By the way, I will be away for the rest of the week, so I expect to continue on this PR starting next week. Looking forward to it.

Reply:
Done.